gf/include/alps/gf/gf_base.hpp

```diff
@@ -185,28 +192,35 @@ namespace alps {
 
         /// construct new green's function from index slice of GF with higher dimension
         template<typename St, typename...OLDMESHES, typename Index, typename ...Indices>
-        gf_base(gf_base<VTYPE, numerics::detail::tensor_base < VTYPE, sizeof...(OLDMESHES), St >, OLDMESHES...> & g, std::tuple<OLDMESHES...>& oldmesh,
-                const mesh_types &meshes, const Index ind, const Indices... idx) :
+        gf_base(gf_base<VTYPE, numerics::detail::tensor_base < VTYPE, sizeof...(OLDMESHES), St >, OLDMESHES...> & g,
+                std::tuple<OLDMESHES...>& oldmesh, const mesh_types &meshes, const Index ind, const Indices... idx) :
           data_(g.data()(ind(), idx()...)), meshes_(meshes), empty_(false) {}
 
         /// construct const view
         template<typename RHSTYPE ,typename St, typename...OLDMESHES, typename ...Indices>
-        gf_base(const gf_base<RHSTYPE, numerics::detail::tensor_base < RHSTYPE, sizeof...(OLDMESHES), St >, OLDMESHES...> & g, const std::tuple<OLDMESHES...>& oldmesh,
-                mesh_types &meshes, const Indices... idx) : data_(g.data()(idx()...)), meshes_(meshes), empty_(false) {}
+        gf_base(const gf_base<RHSTYPE, numerics::detail::tensor_base < RHSTYPE, sizeof...(OLDMESHES), St >, OLDMESHES...> & g,
+                const std::tuple<OLDMESHES...>& oldmesh, mesh_types &meshes, const Indices... idx) :
+            data_(g.data()(idx()...)), meshes_(meshes), empty_(false) {}
 
         /// copy assignment
         gf_type& operator=(const gf_type & rhs) {
-          throw_if_empty();
-          check_meshes(rhs);
+          rhs.throw_if_empty();
+          swap_meshes(rhs.meshes_, make_index_sequence<sizeof...(MESHES)>());
           data_ = rhs.data();
+          empty_= rhs.empty_;
           return *this;
         }
         /// move assignment
-        gf_type& operator=(gf_type && rhs) = default;
+        gf_type& operator=(gf_type && rhs) {
+          swap_meshes(rhs.meshes_, make_index_sequence<sizeof...(MESHES)>());
+          data_ = rhs.data();
+          empty_= rhs.empty_;
```

Comment:
It is possible to move-assign an empty GF into a non-empty one. By the same logic, we may allow to copy-assign an empty GF (see my prev. comment).